pkg/policies/engine/rego/rego.go

```diff
@@ -107,22 +107,28 @@ func (r *Rego) Verify(ctx context.Context, policy *engine.Policy, input []byte,
 	}
 
 	// Try the main rule first
-	if err := executeQuery(mainRule, r.OperatingMode == EnvironmentModeRestrictive); err != nil {
+	if err := executeQuery(resultRule, r.OperatingMode == EnvironmentModeRestrictive); err != nil {
 		return nil, err
 	}
 
 	// If res is nil, it means that the rule hasn't been found
 	if res == nil {
 		// Try with the deprecated main rule
-		if err := executeQuery(deprecatedMainRule, r.OperatingMode == EnvironmentModeRestrictive); err != nil {
+		if err := executeQuery(violationsRule, r.OperatingMode == EnvironmentModeRestrictive); err != nil {
 			return nil, err
 		}
 
 		if res == nil {
-			return nil, fmt.Errorf("failed to evaluate policy: no '%s' nor '%s' rule found", mainRule, deprecatedMainRule)
+			return nil, fmt.Errorf("failed to evaluate policy: neither '%s' nor '%s' rule found", resultRule, violationsRule)
 		}
+
+		return parseViolationsRule(res, policy)
 	}
 
+	return parseResultRule(res, policy)
+}
+
+func parseViolationsRule(res rego.ResultSet, policy *engine.Policy) (*engine.EvaluationResult, error) {
```

Comment:
please add some context about this, why is this best effort. It's important that it's clear in the code this is a migration/compatibility path